hr_employee_seniority/models/hr_employee.py

```diff
@@ -22,29 +22,49 @@ class HrEmployee(models.Model):
         compute='_compute_months_service',
     )
 
-    @api.depends('contract_ids')
+    def _first_contract(self):
+        Contract = self.env['hr.contract']
+        return Contract.search([('employee_id', '=', self.id)],
+                               order='date_start asc', limit=1)
+
+    @api.depends('contract_ids', 'initial_employment_date')
     def _compute_months_service(self):
+        date_now = fields.Date.today()
```

Comment:
@gurneyalex Please see my previous comment, i will try again tomorrow morning

@leemannd I'm not quite sure that this will lead to correct result, tested on runbot and contract from '2016-11-01 to 2016-12-31' will give you only one month of service, from calculation will lead to 12-11=1...which is not correct